test/message_scheduler_test.exs

```diff
@@ -0,0 +1,14 @@
+defmodule Elix.MessageSchedulerTest do
+  use ExUnit.Case
+  doctest Elix.MessageScheduler
+
+  test "sends a message" do
+    assert Elix.MessageScheduler.send_at(now(), :test) == :ok
+
+    # Actually testing this module will be a challenge...
```

Comment:
Part of the trouble is that `assert_receive` asserts that a message was received by *the current process*, which is the test itself in this context. `Elix.MessageScheduler` is hard-coded to send messages to `Elix.Robot`. I could perhaps allow configuring the recipient to `Elix.TestRobot` in tests and registering this test process under that name at the start...